From cced23feb52b51840da1c5a0e1586b82bcfd2062 Mon Sep 17 00:00:00 2001
From: Konstantin Porotchkin <kostap@marvell.com>
Date: Thu, 30 Jun 2016 18:15:06 +0300
Subject: [PATCH 0317/1345] dts: a80x0: Enable PCIe-2 port on CP slave

commit  f366157a492bfef63162666705c7aa4a9dded096 from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

- Add MSI parent information for CP slave PCIe ports
- Enable PCIe-2 port on CP slave

Change-Id: I3a117c831b133e9710536cf50bbddcd561607ff4
Signed-off-by: Konstantin Porotchkin <kostap@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/30877
Reviewed-by: Hanna Hawa <hannah@marvell.com>
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Omri Itach <omrii@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../boot/dts/marvell/armada-8040-db-router.dts     |    3 +++
 arch/arm64/boot/dts/marvell/armada-8040.dtsi       |    9 +++++++++
 2 files changed, 12 insertions(+)

diff --git a/arch/arm64/boot/dts/marvell/armada-8040-db-router.dts b/arch/arm64/boot/dts/marvell/armada-8040-db-router.dts
index b1212a0..7ced92f 100644
--- a/arch/arm64/boot/dts/marvell/armada-8040-db-router.dts
+++ b/arch/arm64/boot/dts/marvell/armada-8040-db-router.dts
@@ -145,5 +145,8 @@
 				};
 			};
 		};
+		pcie@0x640000 {
+			status = "okay";
+		};
 	};
 };
diff --git a/arch/arm64/boot/dts/marvell/armada-8040.dtsi b/arch/arm64/boot/dts/marvell/armada-8040.dtsi
index 99a8de1..9e07cf0 100644
--- a/arch/arm64/boot/dts/marvell/armada-8040.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-8040.dtsi
@@ -175,6 +175,15 @@
 				dma-coherent;
 			};
 		};
+		pcie@0x600000 {
+			msi-parent = <&gic_v2m0>;
+		};
+		pcie@0x620000 {
+			msi-parent = <&gic_v2m0>;
+		};
+		pcie@0x640000 {
+			msi-parent = <&gic_v2m0>;
+		};
 	};
 };
 
-- 
1.7.9.5

